src/main/java/com/google/sps/servlets/EmailServlet.java

```diff
@@ -0,0 +1,160 @@
+      System.out.println("Failed to set email address.");
+    } catch (MessagingException e) {
+      System.out.println("Failed to send email.");
+    } catch (UnsupportedEncodingException e) {
+      System.out.println("Failed to encode email.");
+    }
+  }
+
+  /*
+   * Queries ids of the author of the modified question/answer and its followers.
+   */
+  private List<Integer> getUsersToNotify (String type, int elementId){
+    // Defines the necessary data to access the server.
+    String url = "jdbc:mysql://localhost:3306/Mintern?useSSL=false&serverTimezone=America/Mexico_City";
+    String user = "root";
+    String password = "";
+
+    List<Integer> usersToNotify = new ArrayList<>();
+    if(type.equals("question")) {
+      // If the notification is for an anwer to a question.
+      String query =  "SELECT follower_id FROM QuestionFollower WHERE question_id = " + elementId;
+      // Query the information from QuestionFollower table.
+      try (Connection connection = DriverManager.getConnection(url, user, password);
+           PreparedStatement pst = connection.prepareStatement(query);
+           ResultSet rs = pst.executeQuery()) {
+        while(rs.next()){
+          // Add the current id (first column of ResultSet) to the list.
+          usersToNotify.add(rs.getInt(1));
+        }
+        // Close the connection once the query was performed have been performed.
+        connection.close();
+      } catch (SQLException ex) {
+        Logger lgr = Logger.getLogger(DataServlet.class.getName());
+        lgr.log(Level.SEVERE, ex.getMessage(), ex);
+      }
+    } else if (type.equals("answer")) {
+      // If the notification is for a new comment in an answer.
+      String query =  "SELECT follower_id FROM AnswerFollower WHERE answer_id = " + elementId;
+      // Query the information from AnswerFollower table.
+      try (Connection connection = DriverManager.getConnection(url, user, password);
+           PreparedStatement pst = connection.prepareStatement(query);
+           ResultSet rs = pst.executeQuery()) {
+        while(rs.next()){
+          // Add the current id (first column of ResultSet) to the list.
+          usersToNotify.add(rs.getInt(1));
+        }
+        // Close the connection once the query was performed have been performed.
+        connection.close();
+      } catch (SQLException ex) {
+          Logger lgr = Logger.getLogger(DataServlet.class.getName());
+          lgr.log(Level.SEVERE, ex.getMessage(), ex);
+      }
+    }
+    return usersToNotify;
+  }
+
+  /*
+   * Queries the mails of users to notify and returns them in a single string.
+   */
+  private String getUserEmails (List<Integer> userIds) {
+    // Defines the necessary data to access the server.
+    String url = "jdbc:mysql://localhost:3306/Mintern?useSSL=false&serverTimezone=America/Mexico_City";
+    String user = "root";
+    String password = "";
+
+    String userEmails = new String();
+    for(int userId : userIds){
```

Comment:
nit: Add space before opening parenthesis and add space after closing parenthesis.

Reply:
Done!